fleet_adapter_mir/fleet_adapter_mir/MiRCommandHandle.py

```diff
@@ -493,8 +493,8 @@ def dock_closure():
             while self.rmf_docking_requested:
                 if not self.dry_run:
                     api_response = self.mir_api.status_get()
-                    self.rmf_docking_executed = (
-                        'docking' in api_response['mission_text'].lower())
+                    if ('docking' in api_response['mission_text'].lower()):
```

Comment:
No, the original code is not fine:

In the original code, the value of `self.rmf_docking_executed` is set at every API response. The problem is that on one iteration the response contain "docking" but state is CHARGING. On the next iteration, the response does not contain "docking" and state is READY.

In the previous code, `self.rmf_docking_executed` is set to true when "docking" is in the response, but is then set back to false when the state is READY. Therefor the ending condition for the docking, with both `self.rmf_docking_executed == True` and `STATE == READY` never happens.

With this change, `self.rmf_docking_executed` is set to true once docking has started but is not set back to false. When the robot finishes charging, state becomes READY and both conditions are completed. The robot undock and can receive new missions. 

I hope it's clear

Reply:
I see thanks for explaining that; I get it now.
But that raises another question. What happens when the docking execution is instantaneous or if the `mission_text` changed to `Docking` and the immediate to `Ready` before the next `api_response = self.mir_api.status_get()` call? The while loop would never terminate? Perhaps `response = self.queue_dock_and_charge_mission()` should block till the `Docking` is received or timeout after a while. I don't think this is something that needs to be addressed in this PR but would be good to know if this could happen.

For now, it would be good to document the expectation of docking. Perhaps something like
 ```python
 # Once the robot begins executing a queued docking mission, it's API response's `mission_text` will change to `Docking...` and with `state_text`:`Executing`. But once the docking is completed, the `mission_text` will revert to `Waiting for new missions...` and `state_text`:`Ready...`.
# Below we try to catch the change in `mission_text` to `Docking...` first and then wait till the `state_text` is `Ready...` before considering the docking complete.
 ```

Reply:
> For now, it would be good to document the expectation of docking. Perhaps something like
> 
> ```python
> # Once the robot begins executing a queued docking mission, it's API response's `mission_text` will change to `Docking...` and with `state_text`:`Executing`. But once the docking is completed, the `mission_text` will revert to `Waiting for new missions...` and `state_text`:`Ready...`.
> # Below we try to catch the change in `mission_text` to `Docking...` first and then wait till the `state_text` is `Ready...` before considering the docking complete.
> ```

I added the comments in https://github.com/osrf/fleet_adapter_mir/pull/31/commits/6816f9406ff01ab64cda9139ae66a6856a390411

Reply:
I think we're now calling the function at all the appropriate places. ie, whenever we succeed or fail with executing the docking routine.